### ISSUE TITLE
E0277 does not detected for returned value with`impl %Trait%`

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9411]
## Environment

* **IntelliJ Rust plugin version:** 0.4.180.4901-222-nightly
* **Rust toolchain version:** 1.66.0-nightly (432abd86f 2022-09-20) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.3 EAP (CL-223.4884.72)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
<img width="679" alt="image" src="https://user-images.githubusercontent.com/49211026/192286418-3bcf0717-18da-4c8f-ba05-d6f5089e5e44.png">

## Steps to reproduce
```rust
use std::future::Future;

fn bar() -> impl Future<Output = u8> {
    let x: u8 = 56;
    x + 5
}

```
